src/bidi_class.rs

```diff
@@ -0,0 +1,145 @@
+    }
+
+    // Collect each bidi class into an ordered set.
+    let mut by_type: BTreeMap<String, BTreeSet<u32>> = BTreeMap::new();
+    let mut assigned = BTreeSet::new();
+    for row in rows {
+        assigned.insert(row.codepoint.value());
+        let bc = bidi_class_name(&row.bidi_class)?;
+        by_type
+            .entry(bc)
+            .or_insert(BTreeSet::new())
+            .insert(row.codepoint.value());
+    }
+
+    // Process the codepoints that are not listed as per the notes in
+    // DerivedBidiClass.txt (UCD 12.1):
+    //
+    // Bidi Class (listing UnicodeData.txt, field 4: see UAX #44: http://www.unicode.org/reports/tr44/)
+    // Unlike other properties, unassigned code points in blocks
+    // reserved for right-to-left scripts are given either types R or AL.
+    //
+    // The unassigned code points that default to AL are in the ranges:
+    //     [\u0600-\u07BF \u0860-\u086F \u08A0-\u08FF \uFB50-\uFDCF \uFDF0-\uFDFF \uFE70-\uFEFF
+    //      \U00010D00-\U00010D3F \U00010F30-\U00010F6F
+    //      \U0001EC70-\U0001ECBF \U0001ED00-\U0001ED4F \U0001EE00-\U0001EEFF]
+    //
+    //     This includes code points in the Arabic, Syriac, and Thaana blocks, among others.
+    //
+    // The unassigned code points that default to R are in the ranges:
+    //     [\u0590-\u05FF \u07C0-\u085F \u0870-\u089F \uFB1D-\uFB4F
+    //      \U00010800-\U00010CFF \U00010D40-\U00010F2F \U00010F70-\U00010FFF
+    //      \U0001E800-\U0001EC6F \U0001ECC0-\U0001ECFF \U0001ED50-\U0001EDFF \U0001EF00-\U0001EFFF]
+    //
+    //     This includes code points in the Hebrew, NKo, and Phoenician blocks, among others.
+    //
+    // The unassigned code points that default to ET are in the range:
+    //     [\u20A0-\u20CF]
+    //
+    //     This consists of code points in the Currency Symbols block.
+    //
+    // The unassigned code points that default to BN have one of the following properties:
+    //     Default_Ignorable_Code_Point
+    //     Noncharacter_Code_Point
+    //
+    // For all other cases:
+    //
+    //  All code points not explicitly listed for Bidi_Class
+    //  have the value Left_To_Right (L).
+    #[rustfmt::skip]
+    let default_class_assignments = [
+        (0x0600,0x07BF, "AL"), (0x0860,0x086F, "AL"), (0x08A0,0x08FF, "AL"), (0xFB50,0xFDCF, "AL"),
+        (0xFDF0,0xFDFF, "AL"), (0xFE70,0xFEFF, "AL"), (0x00010D00,0x00010D3F, "AL"),
+        (0x00010F30,0x00010F6F, "AL"), (0x0001EC70,0x0001ECBF, "AL"), (0x0001ED00,0x0001ED4F, "AL"),
+        (0x0001EE00,0x0001EEFF, "AL"),
+
+        (0x0590, 0x05FF, "R"), (0x07C0, 0x085F, "R"), (0x0870, 0x089F, "R"), (0xFB1D, 0xFB4F, "R"),
+        (0x00010800, 0x00010CFF, "R"), (0x00010D40, 0x00010F2F, "R"), (0x00010F70, 0x00010FFF, "R"),
+        (0x0001E800, 0x0001EC6F, "R"), (0x0001ECC0, 0x0001ECFF, "R"), (0x0001ED50, 0x0001EDFF, "R"),
+        (0x0001EF00, 0x0001EFFF, "R"),
+
+        (0x20A0, 0x20CF, "ET")
+    ];
```

Comment:
Perhaps it would be better to move this to a constant? Or perhaps, define it as a constant inside of the `lookup_unassigned` helper function?